sdk/nodejs/npm/yarn.go

```diff
@@ -77,14 +77,17 @@ func (yarn *yarnClassic) Pack(ctx context.Context, dir string, stderr io.Writer)
 	// Since we're planning to read the name of the output from stdout, we create
 	// a substitute buffer to intercept it.
 
-	// generate a new uuid, to be used in the naming of the file.
-	// TODO: We should probably prefer a os.tmpfile instead of recreating
-	// the OS's functionality.
-	uuid, err := uuid.NewV4()
+	// Create a tmpfile to write the tarball to.
+	// It will have the form "pulumi-tarball-12345.tgz", where 12345
+	// is a random string chosen by Go.
+	tmpfile, err := os.CreateTemp("", "pulumi-tarball-*.tgz")
 	if err != nil {
 		return nil, err
 	}
-	packfile := fmt.Sprintf("%s.tgz", uuid.String())
+	packfile := tmpfile.Name()
+	// Clean up the tarball after we're done here.
+	defer os.Remove(packfile)
+	defer tmpfile.Close()
```

Comment:
optional: Consider collapsing this into a single defer since this is all part of a single cleanup

```go
defer func() {
  contract.IgnoreError(tmpfile.Close())
  contract.IgnoreError(os.Remove(packfile))
}()
```

Reply:
Nice! That removes the confusion around LIFO `defer`. <3

Reply:
Done!